cmd/chisel/cmd_cut.go

```diff
@@ -91,11 +91,15 @@ func (cmd *cmdCut) Execute(args []string) error {
 			Suites:     archiveInfo.Suites,
 			Components: archiveInfo.Components,
 			CacheDir:   cache.DefaultDir("chisel"),
+			Pro:        archiveInfo.Pro,
+			Priority:   archiveInfo.Priority,
 		})
 		if err != nil {
 			return err
 		}
-		archives[archiveName] = openArchive
+		if openArchive != nil {
+			archives[archiveName] = openArchive
+		}
```

Comment:
Actually, you are right. Current code doesn't expect non-nil archive when `err` is nil, so this could be a surprise. Applied your suggestion in #60. (`archive.ErrNoProCredentials` is checked in `cmd_cut.go`.)